### PR TITLE
default should map always, not require the minHypervisorVersion

### PR DESCRIPTION
**Pull request by DaanHoogland**
### Description

This PR fixes an logic error in #6758 

### Types of changes

- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Enhancement (improves an existing feature and functionality)
- [ ] Cleanup (Code refactoring and cleanup, that may add test cases)

#### Feature/Enhancement Scale

- [ ] Major
- [ ] Minor

#### Bug Severity

- [ ] BLOCKER
- [ ] Critical
- [ ] Major
- [ ] Minor
- [ ] Trivial

### CODE REVIEW

**Comment by DaanHoogland** [comment]
@blueorangutan package

**Comment by DaanHoogland** [comment]
@blueorangutan test

**Review by weizhouapache — COMMENTED** [review]
code lgtm

However, need to verify if https://github.com/apache/cloudstack/issues/6736 is fixed

**Comment by DaanHoogland** [comment]
@borisstoyanov I think we need this in for the upload of OVAs not having a hypervisor version and preventing those that have a too high hypervisor version.

**Comment by DaanHoogland** [comment]
@blueorangutan test centos7 vmware-67u3

**Comment by weizhouapache** [comment]
code lgtm, but I wonder ##6736 will be back again

**Comment by DaanHoogland** [comment]
> code lgtm, but I wonder ##6736 will be back again

I don´t think so. @borisstoyanov can you look at that?

**Comment by weizhouapache** [comment]
@DaanHoogland 
this is closed ?

**Comment by DaanHoogland** [comment]
> @DaanHoogland this is closed ?

yes it was, forgot to reopen to restart actions...

**Comment by vladimirpetrov** [comment]
From what I can see, this haven't been fixed yet, here is what I got when testing as described in issue #6736:

When registering the mentioned template (PI-VA-3.10.0.0.205.ova) with 'Read from OVA', here is the result:
Did not find a guest OS (Red Hat Enterprise Linux 7 (64-bit)) with type "rhel7_64Guest" and minimal hypervisor hardware version null.

image

when 'Read from OVA' is false, the same template is properly registered.

**Comment by DaanHoogland** [comment]
> From what I can see, this haven't been fixed yet, here is what I got when testing as described in issue #6736:
> 
> When registering the mentioned template (PI-VA-3.10.0.0.205.ova) with 'Read from OVA', here is the result: Did not find a guest OS (Red Hat Enterprise Linux 7 (64-bit)) with type "rhel7_64Guest" and minimal hypervisor hardware version null.
> 
> image
> 
> when 'Read from OVA' is false, the same template is properly registered.

i think we have to check if this is a scenario that would work before this PR.

File: engine/schema/src/main/java/com/cloud/storage/dao/GuestOSHypervisorDaoImpl.java [hunk 1]
```diff
@@ -157,7 +157,7 @@ public List<GuestOSHypervisorVO> listByOsNameAndHypervisorMinimumVersion(String
         sc.and(sc.entity().getGuestOsName(), SearchCriteria.Op.EQ, guestOsName);
         sc.and(sc.entity().getHypervisorType(), SearchCriteria.Op.EQ, hypervisorType);
         sc.and().op(sc.entity().getHypervisorVersion(), SearchCriteria.Op.GTEQ, minHypervisorVersion);
-        sc.or(sc.entity().getHypervisorVersion(), SearchCriteria.Op.NEQ, "default");
+        sc.or(sc.entity().getHypervisorVersion(), SearchCriteria.Op.EQ, "default");
```
Comment (rohityadavcloud) [inline engine/schema/src/main/java/com/cloud/storage/dao/GuestOSHypervisorDaoImpl.java:160]: should this be GTEQ?

Comment (DaanHoogland) [reply engine/schema/src/main/java/com/cloud/storage/dao/GuestOSHypervisorDaoImpl.java:160]: no, default is not part of the simver and it is OR'ed so I think EQ is good

**Comment by weizhouapache** [comment]
added to 4.18.1.0 milestone

**Review by rohityadavcloud — APPROVED** [review]

**Comment by rohityadavcloud** [comment]
Not sure if manual testing is needed, but other LGTM. Thanks @DaanHoogland

**Comment by rohityadavcloud** [comment]
ping @weizhouapache 
@blueorangutan package

**Comment by weizhouapache** [comment]
manual test is required to test if https://github.com/apache/cloudstack/issues/6736 is fixed or not.

**Comment by DaanHoogland** [comment]
@weizhouapache I think this is a solution looking for a problem. Shall I close?

**Comment by weizhouapache** [comment]
> @weizhouapache I think this is a solution looking for a problem. Shall I close?

@DaanHoogland 
The old logic seems wrong but fixed #6736 (the return results are always not empty)
the new logic looks good, but #6736 may come back again (the return results might be null).
I have no idea how to proceed ..

**Comment by DaanHoogland** [comment]
> > @weizhouapache I think this is a solution looking for a problem. Shall I close?
> 
> @DaanHoogland The old logic seems wrong but fixed #6736 (the return results are always not empty) the new logic looks good, but #6736 may come back again (the return results might be null). I have no idea how to proceed ..

ok, closing. we can always re-open

**Pull request closed without merging**